src/core/qgsmaplayer.cpp

```diff
@@ -371,7 +369,34 @@ QString QgsMapLayer::attributionUrl() const
   QGIS_PROTECT_QOBJECT_THREAD_ACCESS_NON_FATAL
 
   return mServerProperties->attributionUrl();
+}
+
+void QgsMapLayer::setLegendUrl( const QString &legendUrl )
+{
+  QGIS_PROTECT_QOBJECT_THREAD_ACCESS_NON_FATAL
```

Comment:
Can we upgrade this to the FATAL error? Or does that break existing tests?

Reply:
Done. Let's see if it breaks anything

Reply:
Besides the grass issue, there is a Hana test failing but I don't think it's related at all to this